### PR TITLE
chromedriver: add dbus to libraries, correct LD_LIBRARY_PATH wrapping

### LINKED ISSUE #139547: Chromedriver can’t find libdbus and crashes
### Describe the bug
`chromedriver --version` crashes with the following message
```
/nix/store/akd5fl2a2xz4rvmhvs347w93yhj2gg1h-chromedriver-94.0.4606.41/bin/chromedriver: error while loading shared libraries: libdbus-1.so.3: cannot open shared object file: No such file or directory
```
(on master, nixpkgs-unstable, nixos-unstable and nixos-21.05)

### Steps To Reproduce
1. `nix-shell --pure -p chromedriver --command 'chromedriver --version'`

### Expected behavior
Chromedriver should report its version.

### Additional context
`git bisect` points towards commit 8d8b451f725b63edec06b033ff2bdaa2f0885b4e (so I guess new version added dependency on dbus)

### Notify maintainers

@cillianderoiste @marsam @primeos   

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.64, NixOS, 21.05.3216.ebf419e737b (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.05.3339.056a3c1fae3"`
 - channels(ivan): `"nixpkgs-unstable-21.11pre317666.1a56d76d718"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- chromedriver
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by ivan-timokhin**
It is apparently required since version 94.

Fixes #139547

###### Motivation for this change

Chromedriver doesn’t run otherwise.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ivan-timokhin** [comment]
Incidentally, a couple of questions:
1. Should I add a test to check that `chromedriver` runs (essentially copy-pasting from [the manual](https://nixos.org/manual/nixpkgs/stable/#sec-package-tests)), to avoid similar accidents in the future?
2. The resulting wrapper script includes the original `LD_LIBRARY_PATH` twice (the new value ends with `:$LD_LIBRARY_PATH'${LD_LIBRARY_PATH:+':'}$LD_LIBRARY_PATH`), so it seems that the explicit reference to `$LD_LIBRARY_PATH` in `wrapProgram` invocation is redundant; should I remove it while I’m here?

If there are no obvious downsides to these changes that I’m oblivious to, should they go into this PR or separate ones?

**Comment by SuperSandro2000** [comment]
> Should I add a test to check that chromedriver runs (essentially copy-pasting from the manual), to avoid similar accidents in the future?

If you want to.

> The resulting wrapper script includes the original LD_LIBRARY_PATH twice (the new value ends with :$LD_LIBRARY_PATH'${LD_LIBRARY_PATH:+':'}$LD_LIBRARY_PATH), so it seems that the explicit reference to $LD_LIBRARY_PATH in wrapProgram invocation is redundant; should I remove it while I’m here?


Yes please, line 50 should be.

```
wrapProgram "$out/bin/chromedriver" --prefix LD_LIBRARY_PATH : "${libs}"
```

**Comment by ivan-timokhin** [comment]
Thanks!

File: pkgs/development/tools/selenium/chromedriver/default.nix [hunk 1]
```diff
@@ -46,9 +48,13 @@ in stdenv.mkDerivation rec {
...
   '' + lib.optionalString (!stdenv.isDarwin) ''
     patchelf --set-interpreter ${glibc.out}/lib/ld-linux-x86-64.so.2 $out/bin/chromedriver
-    wrapProgram "$out/bin/chromedriver" --prefix LD_LIBRARY_PATH : "${libs}:\$LD_LIBRARY_PATH"
+    wrapProgram "$out/bin/chromedriver" --prefix LD_LIBRARY_PATH : "${libs}"
   '';
 
+  passthru.tests = {
+    version-check = callPackage ./tests.nix { };
+  };
+
```
Comment (Artturin) [inline pkgs/development/tools/selenium/chromedriver/default.nix:57]:
```suggestion
  passthru.tests.version =
    testVersion { package = chromedriver; };

```

https://github.com/NixOS/nixpkgs/pull/121896

Comment (ivan-timokhin) [reply pkgs/development/tools/selenium/chromedriver/default.nix]: Thank you!

File: pkgs/development/tools/selenium/chromedriver/default.nix [hunk 2]
```diff
@@ -46,9 +48,12 @@ in stdenv.mkDerivation rec {
     install -m755 -D chromedriver $out/bin/chromedriver
   '' + lib.optionalString (!stdenv.isDarwin) ''
     patchelf --set-interpreter ${glibc.out}/lib/ld-linux-x86-64.so.2 $out/bin/chromedriver
-    wrapProgram "$out/bin/chromedriver" --prefix LD_LIBRARY_PATH : "${libs}:\$LD_LIBRARY_PATH"
+    wrapProgram "$out/bin/chromedriver" --prefix LD_LIBRARY_PATH : "${libs}"
   '';
 
+  passthru.tests.version =
+    testVersion { package = chromedriver; };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/selenium/chromedriver/default.nix:55]:
```suggestion
  passthru.tests.version = testVersion { package = chromedriver; };
```

**Pull request merged**